flexmeasures/ui/views/charts.py

```diff
@@ -0,0 +1,129 @@
+from flask_security import roles_accepted
+from bokeh.embed import json_item
+from marshmallow import Schema, fields
+from webargs.flaskparser import use_args
+from flask_cors import cross_origin
+from flask_json import as_json
+
+from flexmeasures.api.v2_0 import flexmeasures_api as flexmeasures_api_v2_0
+from flexmeasures.api.v2_0.routes import v2_0_service_listing
+from flexmeasures.api.common.schemas import PeriodField
+from flexmeasures.data.queries.analytics import get_power_data
+from flexmeasures.ui.views.analytics import make_power_figure
+
+
+"""
+An endpoint to get a power chart.
+
+This will grow to become code for more charts eventually.
+The plan is to separate charts specs from the actual data later,
+and to switch to Altair.
+
+For now, we'll keep this endpoint here, with route and implementation in the same file.
+When we move forward, we'll review the architecture.
+"""
+
+
+v2_0_service_listing["services"].append(
+    {
+        "name": "POST /charts/power",
+        "access": ["admin", "Prosumer"],
+        "description": "Get a Bokeh chart for power data to embed in web pages.",
+    },
+)
+
+
+class ChartRequestSchema(Schema):
+    """
+    This schema describes the request for a chart.
+    """
+
+    resource = fields.Str(required=True)
+    start_time = fields.DateTime(required=True)
+    end_time = fields.DateTime(required=True)
+    resolution = PeriodField(required=True)
+    show_consumption_as_positive = fields.Bool(missing=True)
+    forecast_horizon = PeriodField(missing="PT6H")
+
+
+@flexmeasures_api_v2_0.route("/charts/power", methods=["POST"])
+@cross_origin()  # origins=["localhost"])
+@roles_accepted("admin", "Prosumer")
```

Comment:
We're mixing USEF roles here (for the first time?), and I'm not sure about its implications. In other endpoints we used a separate decorator to check for USEF roles:

```
@usef_roles_accepted(*list_access(v2_0_service_listing, "POST /charts/power"))
```

Reply:
We used this mix in our main UI views:

```
 grep -r 'admin", "Prosumer' flexmeasures
flexmeasures/ui/views/control.py:@roles_accepted("admin", "Prosumer")
flexmeasures/ui/views/charts.py:        "access": ["admin", "Prosumer"],
flexmeasures/ui/views/charts.py:@roles_accepted("admin", "Prosumer")
flexmeasures/ui/views/portfolio.py:@roles_accepted("admin", "Prosumer")
flexmeasures/ui/views/analytics.py:@roles_accepted("admin", "Prosumer")
flexmeasures/ui/views/analytics.py:@roles_accepted("admin", "Prosumer")
```

Those probably deserve a look, then, as well. If this is wrong, it's a legacy sin.

Reply:
Both implementations (`usef_roles_accepted` and the original `roles_accepted`) are very similar and I don't believe they'd be in each other's way. `usef_roles_accepted` adds checking for the "anonymous"  role and it returns our own `invalid_sender` response instead of routing to an unauth handler. 

I propose to get rid of it soonish. 
In the context of this PR, I don't think I need to do something.